code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -50,19 +52,43 @@
 			to_chat(src, "<span class='userdanger'>With our egg laid, our death approaches rapidly...</span>")
 			addtimer(CALLBACK(src, .proc/death), 100)
 
+/mob/living/simple_animal/hostile/headcrab/death()
+	..()
+	ghostize(TRUE) //you need to be a ghost to be the ling. Here, let me help.
+
 /obj/item/organ/body_egg/changeling_egg
 	name = "changeling egg"
 	desc = "Twitching and disgusting."
 	var/datum/mind/origin
-	var/time
+	var/time = 0
+	var/cooldown = 60
+	var/true_hatch_time = EGG_INCUBATION_TIME
+
+/obj/item/organ/body_egg/changeling_egg/Initialize(mapload)
+	. = ..()
+	var/randomization = (rand(80,120)/100) //+- 20%
+	true_hatch_time = CEILING(true_hatch_time*randomization, 1) 
+
+/obj/item/organ/body_egg/changeling_egg/attack_ghost(mob/dead/observer/user)
+	. = ..()
+	if(isnull(origin) && !QDELETED(user) && user.mind) //you managed to be pulled (IE cloning) right as you attacked.
+		origin = user.mind
+		to_chat(user, "<span class='notice'>You will later spawn as a changeling, please remain as a ghost during this time or you may lose your spot!</span>")
 
 /obj/item/organ/body_egg/changeling_egg/egg_process()
 	// Changeling eggs grow in dead people
 	time++
-	if(time >= EGG_INCUBATION_TIME)
-		Pop()
-		Remove(owner)
-		qdel(src)
+	if(time >= true_hatch_time)
+		if(origin)
+			if(!isobserver(origin.current)) //if you're doing something else when it's time to pop then you lose the role.
+				origin = null
+			else
+				Pop()
+				Remove(owner)
+				qdel(src)
+		else if(time > cooldown)
+			notify_ghosts("A changeling is about to burst! Click to become the changeling!", source = src, action=NOTIFY_ATTACK, flashwindow = FALSE)
```

Comment:
Oversight potential:
- Be Sentient Ling
- Infect
- Oooooo cooler role
- Hop into that
- Other ghost hops in
- They now think they're unbound